### PR TITLE
Bring overlay changes into master

### PR DESCRIPTION
**Pull request by jojimt**
Bringing in changes from the demo branch.

### CODE REVIEW

File: pkg/hostagent/opflex.go [hunk 1]
```diff
@@ -215,7 +214,10 @@ func (agent *HostAgent) updateOpflexConfig() {
 
 	newNodeConfig := agent.discoverHostConfig()
 	if newNodeConfig == nil {
-		panic(errors.New("Node configuration autodiscovery failed"))
+                // FIXME
```
Comment (snaiksat) [inline pkg/hostagent/opflex.go]: Per the "fixme" here, should we not have conditional logic here to only proceed if its "overlay" mode?

**Comment by snaiksat** [comment]
I would request that we first review and merge only the vendor directory changes (commit: 
f787519) in a separate PR. That will reduce the size of the PR for the overlay changes, will help to review better, and much easier to navigate the git history in the future.

**Comment by snaiksat** [comment]
I guess since we are not squashing the commits, the vendor dir related commit will show up in the history after merge as well. But still it will help to have that merged first for the other reasons.

**Comment by jojimt** [comment]
The standard practice is to keep the vendor changes in a commit of its own (not a PR of its own). You can just look at the commit diffs rather than the PR diffs for review.

**Comment by snaiksat** [comment]
The vendor packages that being are adding in this PR are not just specific to this PR, they are used by all the code in this repo.

File: docker/launch-hostagent.sh [hunk 2]
```diff
@@ -37,6 +37,75 @@ if [ -w ${PREFIX} ]; then
...
+}
+
+function get_ip {
+    ip addr show "$1" | grep -o 'inet [0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+' | grep -o [0-9].*
+}
+
+function get_mac {
+    ip link show "$1" | awk '/ether/ {print $2}'
+}
+
```
Comment (snaiksat) [inline docker/launch-hostagent.sh]: L52-108 should be done conditionally in the case of overlay mode.

File: cmd/crd-example/allall.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+apiVersion: aci.aw/v1
```
Comment (snaiksat) [inline cmd/crd-example/allall.yaml]: What does .aw stand for?

File: docker/launch-hostagent.sh (hunk 2 above)
Comment (jojimt) [reply docker/launch-hostagent.sh]: This is intentional per @mandeepdhami's ask. The benefit is that we can perform health checks locally.

File: cmd/crd-example/allall.yaml (hunk 3 above)
Comment (jojimt) [reply cmd/crd-example/allall.yaml]: AW as in AnyWhere. But, we could change this as appropriate later.

File: docker/launch-hostagent.sh (hunk 2 above)
Comment (jojimt) [reply docker/launch-hostagent.sh]: Good point. We should take it from the config.

**Comment by jojimt** [comment]
@snaiksat this now excludes the vendor changes.

**Comment by snaiksat** [comment]
@jojimt Great, thanks!

File: docker/Dockerfile-gbpserver [hunk 4]
```diff
@@ -0,0 +1,14 @@
+FROM alpine:3.7
```
Comment (snaiksat) [inline docker/Dockerfile-gbpserver]: At some point we should consider probably move this to 3.9 (our other base images are all 3.9).

File: docker/Dockerfile-opflex-build [hunk 5]
```diff
@@ -1,5 +1,5 @@
-FROM noiro/opflex-build-base
-ARG BUILDOPTS="--with-static-boost --with-boost=/usr/local/boost_1_58_0"
+FROM challa/opflex-build-base:demo
+ARG BUILDOPTS=
```
Comment (snaiksat) [inline docker/Dockerfile-opflex-build]: Can we go back to using noiro/opflex-build-base with the static linking (that is how we are shipping today)?

File: docker/Dockerfile-opflex-build-base [hunk 6]
```diff
@@ -19,12 +19,3 @@ RUN git clone https://github.com/openvswitch/ovs.git --branch v2.6.0 --depth 1 \
...
   && find lib -name "*.h" -exec cp --parents {} "$ROOT/include/openvswitch/" \;
-RUN wget http://10.30.120.20:8000/boost_1_58_0.tar.gz \
-  && tar zxvf boost_1_58_0.tar.gz \
-  && cd boost_1_58_0 \
-  && ./bootstrap.sh --prefix=$ROOT/boost_1_58_0 \
-  && ./b2 cxxflags=-fPIC cflags=-fPIC -a \
-  && ./b2 install --prefix=$ROOT/boost_1_58_0 \
-  && cd .. \
-  && rm -Rf boost_1_58_0.tar.gz \
-  && rm -Rf boost_1_58_0 \;
```
Comment (snaiksat) [inline docker/Dockerfile-opflex-build-base]: Per earlier question/comment, can we revert this?

File: docker/launch-hostagent.sh [hunk 7]
```diff
@@ -38,6 +38,83 @@ if [ -w ${PREFIX} ]; then
...
+    VTEP_IFACE=eth0
+elif check_eth enp0s8; then
+    VTEP_IP=$(get_ip enp0s8)
+    VTEP_IFACE=enp0s8
+else
+    echo "VTEP interface not found"
+fi
+
+# FIXME make route addition based on pod subnet.
+CHECK=$(ip route show 10.2.56.0/24 | wc -l)
```
Comment (snaiksat) [inline docker/launch-hostagent.sh]: Is there a plan to fix this hard coded value (here and below)?

File: docker/launch-opflexagent.sh [hunk 8]
```diff
@@ -27,7 +27,7 @@ if [ -d ${OPFLEXAGENT_CONF_PATH} ]; then
         "name": "disabled",
         "domain": "disabled",
         "ssl": {
-            "mode": "encrypted",
+            "mode": "disabled",
```
Comment (snaiksat) [inline docker/launch-opflexagent.sh]: At least for the APIC solution this needs to be encrypted (as before).

File: pkg/loadbalancer/aws.go [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+	}
+
+	return add, del
+}
+
+//var instanceMap = map[string]string{
+//	"10.2.1.190": "i-0059ba93bf47150cc",
+//	"10.2.1.205": "i-045ce01d1eee099c2",
+//	"10.2.1.240": "i-065531517fe9ba527",
+//}
```
Comment (snaiksat) [inline pkg/loadbalancer/aws.go]: nit: remove if not used

File: docker/Dockerfile-opflex-build (hunk 5 above)
Comment (mchalla) [reply docker/Dockerfile-opflex-build]: Reverted

File: docker/Dockerfile-opflex-build-base (hunk 6 above)
Comment (mchalla) [reply docker/Dockerfile-opflex-build-base]: Done

File: docker/launch-hostagent.sh (hunk 7 above)
Comment (mchalla) [reply docker/launch-hostagent.sh]: fixed.

**Comment by jojimt** [comment]
@snaiksat I believe all review comments have been addressed. Let me know if you have any other concerns before we can merge this.

File: pkg/apicapi/apic_metadata.go [hunk 10]
```diff
@@ -142,6 +142,166 @@ func injectedSvcPortNormalizer(b *ApicObjectBody) {
...
+			"name":      "",
+			"nameAlias": "",
+		},
+		children: []string{
+			"fvRsBd",
+			"fvRsCons",
+			"fvRsProv",
+		},
+	},
+	"cloudEPg": {
```
Comment (kishorejonnala) [inline pkg/apicapi/apic_metadata.go]: cloudEPg shall have relationship to the VRF - "fvRsCtx"

File: pkg/apicapi/apicapi.go [hunk 11]
```diff
@@ -783,6 +795,102 @@ func (conn *ApicConnection) queueDn(dn string) {
...
+	}
+
+	complete(resp)
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Staus: %v", resp.StatusCode)
+	}
+	return nil
+}
+
+func (conn *ApicConnection) PostDnInline(dn string, obj ApicObject) error {
```
Comment (kishorejonnala) [inline pkg/apicapi/apicapi.go]: It will good if we can avoid code duplication in PostTestAPI , postDnInline

**Pull request merged**